### PR TITLE
refactor: use `-night` to deduplicate browser flag colors

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Purpose/Description

organize a little the mess that the themes are

## How Has This Been Tested?

[colors.webm](https://github.com/ankidroid/Anki-Android/assets/69634269/e5d25f72-f16d-45ea-a31d-9f19c2c1145f)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 1]
```diff
@@ -1961,27 +1962,30 @@ open class CardBrowser :
...
-                        } else if (card.queue == Consts.QUEUE_TYPE_SUSPENDED) {
-                            R.attr.suspendedColor
-                        } else {
-                            android.R.attr.colorBackground
-                        }
-                    }
-                }
+        @ColorInt
+        fun getBackgroundColor(context: Context): Int {
+            val flagColor = when (card.userFlag()) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1967]: This would be better extracted to `Flag`

**Review by david-allison — APPROVED** [review]
Looks good! Comment is optional

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt (hunk 1 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1967]: What would be the background color of `Flag.NONE`?

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1967]: Just 1967...1976, `NONE` -> `null`

You're more in the area, so if you feel it's not worthwhile, let's leave it

**Review by lukstbit — APPROVED** [review]

**Pull request merged**